src/hooks/on-create.ts

```diff
@@ -0,0 +1,24 @@
+import { readFileSync, writeFileSync } from 'fs'
+import path from 'path'
+
+type Hook = (args: { projectName: string; directoryPath: string }) => void
+
+// <templateName, Hook[]>
+export const ON_CREATE_HOOKS = new Map<string, Hook[]>()
+
+export const addOnCreateHook = (templateName: string, hook: Hook) => {
+  const hooks = ON_CREATE_HOOKS.get(templateName) || []
+  hooks.push(hook)
+  ON_CREATE_HOOKS.set(templateName, hooks)
+}
+
+const PROJECT_NAME_REPLACE_KEY = '[DYNAMIC_PROJECT_NAME]'
```

Comment:
I like the usage of `%%PROJECT_NAME%%` as it is used in general template engines.